PipeMethodCalls/Endpoints/PipeServer.cs

```diff
@@ -32,6 +32,25 @@ public PipeServer(string pipeName, Func<THandling> handlerFactoryFunc, PipeOptio
 			this.handlerFactoryFunc = handlerFactoryFunc;
 			this.options = options;
 		}
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="PipeServer{THandling}"/> class.
+		/// </summary>
+		/// <param name="rawPipe">Raw pipe stream to wrap with method call capability.</param>
+		/// <param name="handlerFactoryFunc">A factory function to provide the handler implementation.</param>
+		/// <exception cref="ArgumentException">Provided pipe cannot be wrapped. Provided pipe must be setup with the following: PipeDirection - <see cref="PipeDirection.InOut"/>,  PipeTransmissionMode - <see cref="PipeTransmissionMode.Byte"/>, and PipeOptions - <see cref="PipeOptions.Asynchronous"/></exception>
+		public PipeServer(NamedPipeServerStream rawPipe, Func<THandling> handlerFactoryFunc)
+		{
+			if (rawPipe.CanRead && rawPipe.CanWrite && rawPipe.IsAsync && rawPipe.TransmissionMode == PipeTransmissionMode.Byte)

```

Comment:
CanRead, CanWrite, IsAsync and TransmissionMode are all inherited from PipeStream so you could pull this checking logic out into a common function for use on clients and servers.

Reply:
Good catch, that's a lot cleaner.  I apologize for missing that in the first pass.  I will try to get an update to the PR in, tomorrow.  Thanks!